### PR TITLE
Introduce rollup to bundle the package

### PR DESCRIPTION
**Pull request by HarelM**
I've tested it locally with maplibre-gl-js project and it seems to work as expected.
Let me know if there are more changes that need to be done.
Here's the generated package if you want to try it out locally.
[junit-report-builder-4.0.0.zip.zip](https://github.com/user-attachments/files/16426962/junit-report-builder-4.0.0.zip.zip)

### CODE REVIEW

**Review by SimeonC — APPROVED** [review]
Looks good to me, I can double check the output at work tomorrow but from reading here it should be fine. Thanks for raising the PR!

**Comment by davidparsson** [comment]
Thanks, that's quick! I'll try to review and try it out in a few hours. If everything is fine I'll release 4.0.1 and deprecate 4.0.0.

**Comment by davidparsson** [comment]
Oh, I realised that I was unclear. I'll hold again, and await the testing by @SimeonC before releasing.

**Comment by SimeonC** [comment]
Looks good to me!

File: src/index.ts [hunk 1]
```diff
@@ -2,3 +2,7 @@ import { Factory } from './factory';
 
 export default new Factory().newBuilder();
 export type { JUnitReportBuilder as Builder } from './builder';
+export type { TestSuite } from './test_suite';
+export type { TestCase } from './test_case';
+export type { TestGroup } from './test_group';
```
Comment (davidparsson) [inline src/index.ts:7]: If I recall correctly, `TestGroup` could/should be `abstract`, and if so it should not be exported here. But I'll confirm and change that myself.

File: tsconfig.json [hunk 2]
```diff
@@ -1,13 +1,11 @@
 {
   "compilerOptions": {
     "target": "es2018",
-    "module": "commonjs",
+    "module": "ESNext",
```
Comment (davidparsson) [inline tsconfig.json:4]: Is `ESNext` what Rollup expects, or does that introduce any constraints on supported node.js versions? Or does it not matter, as long as the TypeScript source files don't use any shiny new features?

**Review by davidparsson — APPROVED** [review]

File: package.json [hunk 3]
```diff
@@ -70,9 +70,7 @@
     }
   },
   "files": [
-    "./dist/*",
-    "README.md"
```
Comment (davidparsson) [inline package.json]: Why is this not needed? Is this included anyway?

File: tsconfig.json (hunk 2 above)
Comment (HarelM) [reply tsconfig.json:4]: This is what rollup needs.

Comment (HarelM) [reply tsconfig.json:4]: Target allows to "dumb down" features.

File: package.json (hunk 3 above)
Comment (HarelM) [reply package.json]: Readme is not needed, dist is.

**Comment by davidparsson** [comment]
Brilliant, thanks again!

I'll run some more tests, update the readme a bit more, and then get a new release out.

**Pull request merged**